keycloak-common/src/main/java/eu/europeana/keycloak/timer/FixedRateTaskScheduler.java

```diff
@@ -0,0 +1,77 @@
+package eu.europeana.keycloak.timer;
+
+import java.time.LocalDateTime;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+import org.jboss.logging.Logger;
+import org.keycloak.models.KeycloakSessionFactory;
+import org.keycloak.services.scheduled.ClusterAwareScheduledTaskRunner;
+
+/**
+ * Scheduler for running the tasks which are aligned to clock minute which is important for
+ * executions on the clustered environment to avoid task being run at different times on different pods. *
+ */
+
+public class FixedRateTaskScheduler {
+
+  private static final Logger LOG = Logger.getLogger(FixedRateTaskScheduler.class);
+  public static final int MINUTES_IN_HOUR = 60;
+  public static final long MILLISECONDS_IN_A_MINUTE = 6000L;
+  private final ScheduledExecutorService scheduler = Executors.newSingleThreadScheduledExecutor();
+  private final CustomScheduledTask task;
+  private final int intervalMinutes;
+
+  /**
+   * Initialize FixedRateTaskScheduler with task and interval
+   * @param task to be scheduled
+   * @param intervalMinutes between 2 consecutive task executions
+   */
+  public FixedRateTaskScheduler(CustomScheduledTask task, int intervalMinutes) {
+    this.task = task;
+    this.intervalMinutes = intervalMinutes;
+  }
+
+  /**
+   * Schedule the task with delay.
+   * @param keycloakSessionFactory required by {@code ClusterAwareScheduledTaskRunner}
+   */
+  public void scheduleTask(KeycloakSessionFactory keycloakSessionFactory){
+    LOG.info("Scheduling the task '"+task.getTaskName()+"' with interval "+intervalMinutes);
```

Comment:
I've seen a lot of string concatenation when generating logs. Generally speaking it's more efficient to use placeholders when generating logs, e.g. `LOG.info("Scheduling the task {}, task.getTaskName());`. In this case the task.getTaskName() method not called when the log level for this class is higher than info, whereas with concatenation it is always called.
We used to have a Sonar warning for this, but for some reason I don't see that anymore.